addons/im_livechat/models/chatbot_script_step.py

```diff
@@ -0,0 +1,334 @@
+                'phone': input_phone,
+            })
+        elif not self.env.user._is_public():
+            partner = self.env.user.partner_id
+            if update_partner:
+                # update email/phone value from partner if not set
+                update_values = {}
+                if input_email and not partner.email:
+                    update_values['email'] = input_email
+                if input_phone and not partner.phone:
+                    update_values['phone'] = input_phone
+                if update_values:
+                    partner.write(update_values)
+
+        description = Markup('')
+        if input_email:
+            description += Markup('%s<strong>%s</strong><br>') % (_('Please contact me on: '), input_email)
+        if input_phone:
+            description += Markup('%s<strong>%s</strong><br>') % (_('Please call me on: '), input_phone)
+        if description:
+            description += Markup('<br>')
+
+        return {
+            'partner': partner,
+            'email': input_email,
+            'phone': input_phone,
+            'description': description,
+        }
+
+    def _fetch_next_step(self, selected_answer_ids):
+        """ Fetch the next step depending on the user's selected answers.
+            If a step contains multiple triggering answers from the same step the condition between
+            them must be a 'OR'. If is contains multiple triggering answers from different steps the
+            condition between them must be a 'AND'.
+
+            e.g:
+
+            STEP 1 : A B
+            STEP 2 : C D
+            STEP 3 : E
+            STEP 4 ONLY IF A B C E
+
+            Scenario 1 (A C E):
+
+            A in (A B) -> OK
+            C in (C)   -> OK
+            E in (E)   -> OK
+
+            -> OK
+
+            Scenario 2 (B D E):
+
+            B in (A B) -> OK
+            D in (C)   -> NOK
+            E in (E)   -> OK
+
+            -> NOK
+        """
+        self.ensure_one()
+        domain = [('chatbot_script_id', '=', self.chatbot_script_id.id), ('sequence', '>', self.sequence)]
+        if selected_answer_ids:
+            domain = expression.AND([domain, [
+                '|',
+                ('triggering_answer_ids', '=', False),
+                ('triggering_answer_ids', 'in', selected_answer_ids.ids)]])
+        steps = self.env['chatbot.script.step'].search(domain)
+        for step in steps:
+            if not step.triggering_answer_ids:
+                return step
+            answers_by_step = defaultdict(list)
+            for answer in step.triggering_answer_ids:
+                answers_by_step[answer.script_step_id.id].append(answer)
+            if all(any(answer in step_triggering_answers for answer in selected_answer_ids)
+                   for step_triggering_answers in answers_by_step.values()):
+                return step
+        return self.env['chatbot.script.step']
+
+    def _get_forward_operator_parent(self, all_steps):
```

Comment:
I don't think the name is linked to what it does, forward operator is not really something linked to this method. It seems it is a "parent or brother (the 'all part)" based on triggering answer ids ?